app/src/sharedTest/java/org/oppia/app/onboarding/OnboardingFragmentTest.kt

```diff
@@ -0,0 +1,486 @@
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.CoreMatchers.allOf
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.CoreMatchers.not
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.profile.ProfileActivity
+
+/** Tests for [OnboardingFragment]. */
+@RunWith(AndroidJUnit4::class)
+class OnboardingFragmentTest {
+
+  @Before
+  fun setUp() {
+    Intents.init()
+  }
+
+  @Test
+  fun testOnboardingFragment_checkDefaultSlideTitle_isCorrect() {
+    launch(OnboardingActivity::class.java).use {
+      onView(
+        allOf(
+          withId(R.id.slide_title_text_view),
+          isCompletelyDisplayed()
+        )
+      ).check(matches(withText(R.string.slide_0_title)))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_checkDefaultSlideDescription_isCorrect() {
+    launch(OnboardingActivity::class.java).use {
+      onView(
+        allOf(
+          withId(R.id.slide_description_text_view),
+          isCompletelyDisplayed()
+        )
+      ).check(matches(withText(R.string.slide_0_description)))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_checkDefaultSlideImage_isCorrect() {
+    launch(OnboardingActivity::class.java).use {
+      onView(
+        allOf(
+          withId(R.id.slide_image_view),
+          isCompletelyDisplayed()
+        )
+      ).check(matches(withTagValue(equalTo(R.drawable.ic_onboarding_0))))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_checkDefaultSlide_index0DotIsActive_otherDotsAreInactive() {
+    launch(OnboardingActivity::class.java).use {
+      onView(
+        allOf(
+          withId(R.id.onboarding_dot_0),
+          isCompletelyDisplayed()
+        )
+      ).check(matches(withAlpha(1.0F)))
+      onView(
+        allOf(
+          withId(R.id.onboarding_dot_1),
+          isCompletelyDisplayed()
+        )
+      ).check(matches(withAlpha(0.3F)))
+      onView(
+        allOf(
+          withId(R.id.onboarding_dot_2),
+          isCompletelyDisplayed()
+        )
+      ).check(matches(withAlpha(0.3F)))
+      onView(
+        allOf(
+          withId(R.id.onboarding_dot_3),
+          isCompletelyDisplayed()
+        )
+      ).check(matches(withAlpha(0.3F)))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_checkDefaultSlide_skipButtonIsVisible() {
+    launch(OnboardingActivity::class.java).use {
+      onView(withId(R.id.skip_text_view)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_checkDefaultSlide_clickSkipButton_shiftsToLastSlide() {
+    launch(OnboardingActivity::class.java).use {
+      onView(withId(R.id.skip_text_view)).perform(click())
+      SystemClock.sleep(500)
+      onView(
+        allOf(
+          withId(R.id.slide_title_text_view),
+          isCompletelyDisplayed()
```

Comment:
Why again do we need to check that they are completely displayed?

Reply:
`isDisplayed` return true even in if the view is in background and is not visible completely on the screen, whereas `isCompletelyDisplayed` makes sure that the view is 100% visible on the screen.

I can give you more information on this with correct test cases once this animation related issue is resolved because currently it is hard to give a perfect example for the above question as test cases are failing because of this waiting thing also.